unstable test gtid

### Which jobs are flaking?

gtid

### Which test(s) are flaking?

DM-IT

### Jenkins logs or GitHub Actions link

https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/7280/pipeline

```
[2022-06-28T08:19:47.127Z] check diff failed 20-th time, retry later

[2022-06-28T08:19:49.025Z] check diff failed at last
```

```
[2022-06-28T08:19:49.838Z] [2022/06/28 16:19:06.457 +08:00] [ERROR] [subtask.go:377] ["unit process error"] [subtask=test] [unit=Sync] ["error information"="ErrCode:11121 ErrClass:\"functional\" ErrScope:\"internal\" ErrLevel:\"high\" Message:\"fail to restart streamer controller: no relay pos match gtid 05279c58-f6ba-11ec-b2b6-261c1a5f24f8:1-3\" "]
```

gtid test turned off auto-resume

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

```
[2022-07-21T02:38:30.904Z] [2022/07/21 10:37:47.306 +08:00] [ERROR] [subtask.go:377] ["unit process error"] [subtask=test] [unit=Sync] ["error information"="ErrCode:11121 ErrClass:\"functional\" ErrScope:\"internal\" ErrLevel:\"high\" Message:\"fail to restart streamer controller: no relay pos match gtid 53c6992e-089d-11ed-8027-4e905788d272:1-3\" "]
``` 

caused by we disable auto-resume in this test case.

--- a/dm/tests/gtid/run.sh
+++ b/dm/tests/gtid/run.sh
@@ -109,7 +109,7 @@ function run() {
 	run_sql_source2 "insert into gtid.t2 values (4)"
 	# now Previous_gtids event is 09bec856-ba95-11ea-850a-58f2b4af5188:1-4:6
 
-	sleep 1
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
 	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
 		"stop-task test" \
 		"\"result\": true" 3

